### PR TITLE
LANG_007a updates and improvements [KK,LA]

### LINKED ISSUE #80: LANG_007a_A Testing GUI Overview
- limit the selection possibility 

Languages must not be mixed, so you should limit the selection regarding the source and target language. A list from VocabularySelectionModel should be passed to train and study. Before you do this, you should check if something has been selected at all and give meaningful feedback to the user.

### PR DESCRIPTION
**Pull request by alc26**

### CODE REVIEW

File: src/main/java/at/tugraz/sw20asd/lang/ui/Controller/AddVocab.java [hunk 1]
```diff
@@ -79,13 +79,14 @@ public AddVocab(VocabularyAccess vocab) {
     public void initialize() {
         user_info.setVisible(false);
         ObservableList<String> languages =
-                FXCollections.observableArrayList("German", "English");
+                FXCollections.observableArrayList("German", "English", "Chinese");
```
Comment (M-Schiller) [inline src/main/java/at/tugraz/sw20asd/lang/ui/Controller/AddVocab.java]: Please remove this changes from this PR as it is part of LANG_015.

File: src/main/java/at/tugraz/sw20asd/lang/ui/Controller/AddVocab.java [hunk 2]
```diff
@@ -79,13 +79,14 @@ public AddVocab(VocabularyAccess vocab) {
...
 
         from_choice.setItems(languages);
         to_choice.setItems(languages);
 
         language_map.put("German", Locale.GERMAN);
         language_map.put("English", Locale.ENGLISH);
+        language_map.put("Chinese", Locale.CHINESE);
```
Comment (M-Schiller) [inline src/main/java/at/tugraz/sw20asd/lang/ui/Controller/AddVocab.java]: See prior comment regarding LANG_015.

File: src/main/java/at/tugraz/sw20asd/lang/ui/models/VocabularySelectionModel.java [hunk 3]
```diff
@@ -4,26 +4,65 @@
 import javafx.beans.property.SimpleBooleanProperty;
 import javafx.beans.property.SimpleLongProperty;
 import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
```
Comment (M-Schiller) [inline src/main/java/at/tugraz/sw20asd/lang/ui/models/VocabularySelectionModel.java]: This import seems to be unnecessary.

File: src/main/java/at/tugraz/sw20asd/lang/ui/models/VocabularySelectionModel.java [hunk 4]
```diff
@@ -4,26 +4,65 @@
...
+
     private final SimpleBooleanProperty selected;
 
+
     public static VocabularySelectionModel fromDto(VocabularyBaseDto v) {
         VocabularySelectionModel vsm = new VocabularySelectionModel();
 
         vsm.setVocabularyId(v.getId());
         vsm.setVocabularyName(v.getName());
+        vsm.setVocabularySrc(v.getSourceLanguage().toString().toUpperCase() + " - " +
```
Comment (M-Schiller) [inline src/main/java/at/tugraz/sw20asd/lang/ui/models/VocabularySelectionModel.java]: Here, the contents of line 24 and line 27 seem to be mixed up.

File: src/main/java/at/tugraz/sw20asd/lang/ui/Controller/SelectionOverview.java [hunk 5]
```diff
@@ -81,13 +85,23 @@ public void handle(ActionEvent event) {
         study_btn.setOnAction(new EventHandler<ActionEvent>() {
             public void handle(ActionEvent event) {
                 getSelection();
+                if (selectedItems.size() == 0) {
```
Comment (M-Schiller) [inline src/main/java/at/tugraz/sw20asd/lang/ui/Controller/SelectionOverview.java]: To better express your intent, I would prefer if you used `isEmpty()` for checks of emptiness on classes that support it.

File: src/main/java/at/tugraz/sw20asd/lang/ui/Controller/SelectionOverview.java [hunk 6]
```diff
@@ -116,13 +130,18 @@ private void getVocabGroups() {
                         updateUserInformation("no_vocabs");
                     });
                 } else {
+
+                    possibleItems = (FXCollections.observableArrayList(
```
Comment (M-Schiller) [inline src/main/java/at/tugraz/sw20asd/lang/ui/Controller/SelectionOverview.java]: Is there any particular reason, why the RHS part is encompassed by parentheses?

File: LANG_007a.eml [hunk 7]
```diff
@@ -0,0 +1,154 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (M-Schiller) [inline LANG_007a.eml]: Whatever this file is, please do not add it.

**Review by M-Schiller — CHANGES_REQUESTED** [review]
Please remove the *.userlibraries file and the *.eml file and take a look at the suggestions.

File: src/main/java/at/tugraz/sw20asd/lang/ui/Controller/AddVocab.java (hunk 1 above)
Comment (alc26) [reply src/main/java/at/tugraz/sw20asd/lang/ui/Controller/AddVocab.java]: Solve it.

**Review by M-Schiller — APPROVED** [review]
This looks great now.

**Pull request merged**